### DISCUSSION TITLE
How to use the "Configuration tool" to publish/send readings to MQTT server on my network

### DISCUSSION THREAD

**Discussion by fritserasmus** [SBFspot/sbfspot-config#59] [Q&A]
I have a working SBF-Spot installation running on an RPi 3B

I was looking for a solution to get the readings on my local installed MQTT server and found this article:
"SBFspot SMA inverter, MQTT and wrong energy dashboard values"
[https://community.home-assistant.io/t/sbfspot-sma-inverter-mqtt-and-wrong-energy-dashboard-values/446679]([https://community.home-assistant.io/t/sbfspot-sma-inverter-mqtt-and-wrong-energy-dashboard-values/446679])
For now, I am **not worried about the "wrong energy values"**!

**What I need to know:**
1. How do I see what version of SBFspot runs on my RPi?
2. How do I go about just doing an upgrade to the latest version so I retain values?
3. Is the "Configuration tool" as on github "Installation/Configuration tool for SBFspot on Raspberry Pi" another required software OR is it standard on all SBFspot installations?
4. Would it be better to rather redo a fresh install using the latest version?
5. Where can I find some pointers on how to publish values to an MQTT server on my network?

I will really appreciate some pointers so I can get going with this.

Thanks for helping me.

**Comment by sbf-** [comment]
>How do I see what version of SBFspot runs on my RPi?

SBFspot -?

>How do I go about just doing an upgrade to the latest version so I retain values?

Untar the release of your choice and replace SBFspot executable

>Is the "Configuration tool" as on github "Installation/Configuration tool for SBFspot on Raspberry Pi" another required software OR is it standard on all SBFspot installations?

It's just a script to ease the installation and configuration of SBFspot. Not required nor standard.

>Would it be better to rather redo a fresh install using the latest version?

A fresh install will use existing configurations and add the new items like MQTT, so in most cases it is the recommended way to go.

>Where can I find some pointers on how to publish values to an MQTT server on my network?

See [wiki](https://github.com/SBFspot/SBFspot/wiki/MQTT)

**Reply by fritserasmus** [reply]
Thanks a million for your response - I do appreciate it very much!


> Untar the release of your choice and replace SBFspot executable

Should I UNTAR this?

[https://github.com/SBFspot/SBFspot/releases](url)

**Comment by fritserasmus** [comment]
SBFspot-? - Not working for me
`pi@PVOutput:~ $ SBFspot -?

-bash: SBFspot: command not found
`

**Comment by fritserasmus** [comment]
I opted for a completely new re-build of my SBFspot server.

I followed the instructions to the T - really easy to do though, but now I only have ONE SunnyBoy configured not TWO.

Could you please be so kind as to check out my question here:-
https://github.com/SBFspot/sbfspot-config/discussions/60

and here:-
https://forum.pvoutput.org/t/how-do-i-configure-sbfspot-cfg-for-mis-multi-inverter-support/6252

**Reply by sbf-** [reply]
No need to cross-post between sites...
I do my best to answer questions when I have time ;-)
